### ISSUE TITLE
honoring legacy environment variable `NODE`.  Please switch to using `EM_NODE_JS` instead`, unable to switch to EM_NODE_JS

### ISSUE THREAD

**Issue by PaulNewling** [emscripten-core/emsdk#862]
Hello,

I am able to run 'emcc main.cpp -s EXPORTED_FUNCTIONS="['_main']" -o output.js' via my terminal, but when run locally on a node v14.15.5 server via an execSync() call:

```
 const cmdLine = `emcc ${fileName} -s EXPORTED_FUNCTIONS="['_main']" -o output.js`;
           // Run the command line in the folder that was created
           execSync(
             cmdLine,
             {
               cwd: path.join(__dirname, `/tempData/${workingFolder}/`),
             },
             (error, stdout, stderr) => {
               if (error) {
                 console.log(`error: ${error.message}`);
                 return;
               }
               if (stderr) {
                 console.log(`stderr: ${stderr}`);
                 return;               }
               console.log(`stdout: ${stdout}`);             }
           );
```

It outputs the warning in the warning ```warning: honoring legacy environment variable `NODE`.  Please switch to using `EM_NODE_JS` instead``` and the server crashes.

I've been unable to locate any documentation on how to switch to EM_NODE_JS. I assume this would need to be added to the `.emscripten` file, but I unsure what this entails as I am very new to wasm/emscripten.

Any assistance would be much appreciated.

**Comment by kripken** [comment]
I believe those are environment vars. However, the .emscripten file does map some of those, and I'm actually not sure precisely which. Is that documented @sbc100 ?

**Comment by sbc100** [comment]
There seem be a couple of different issues here.   Firstly I seem that somebody (perhaps nodejs itself is setting the `NODE` env var).

Secondly, that is just a warning and should not cause anything to actually fail, which makes me think there is something else going on?  What do you mean exactly by "server crashes."?

**Comment by sbc100** [comment]
I confirmed that that node itself does not set `NODE` and I can run (a slightly modified) version of the script above without generating that warning.

This means that some other part of your system is setting `NODE` in your environment.   You can confirm by running `echo $NODE` or `env | grep NODE` to see this. 

However, as I said above the warning should be harmless enough and should not cause emcc to fail, so there must be something else going on.

**Comment by PaulNewling** [comment]
So it turns out that it was something else in my code that was causing my server. (This was my mistake I'm still pretty new.)

It turns out that the code was executing and compiling, I and another function that was still being called that was deleting the folder in the temp folder that I thought was not getting called, but actually was.

> This means that some other part of your system is setting NODE in your environment. You can confirm by running echo $NODE or env | grep NODE to see this.

I will follow up on this advice. Thank you all for the response it appears my inexperience created my own problems.

**Comment by Taytay** [comment]
I just ran into this exact issue compiling sql.js. I couldn't for the life of me figure out what was setting the NODE environment. As it turns out, we run our build scripts using `npm run build`, and [using `npm run` does in fact set the NODE environment variable when it runs](https://docs.npmjs.com/cli/v6/commands/npm-run-script). So, to get around this, I am now setting EM_NODE_JS in our devcontainer's `/etc/bash.bashrc` so that it's honored:
`export EM_NODE_JS="$EMSDK_NODE"' >> /etc/bash.bashrc`

We do that because EMSDK_NODE is already configured in our devcontainer. We copied this part of the EMSDK's Dockerfile: 
https://github.com/emscripten-core/emsdk/blob/894c25b52ae42ca619059e12dbf88c4a3a62f2c5/docker/Dockerfile#L66-L69

I'm including this comment here for two reasons: 
1) To help other unfortunate folks who run emsdk through `npm run`
2) To ensure that my fix is sound. I now have all of the following ENV variables set to the same value: `EM_NODE_JS`, `EMSDK_NODE`, and `NODE`.
I don't think that `EMSDK_NODE` is respected by the same code complaining about the `NODE` environment variable, but I just wanted to make sure.
